cpp/ql/test/experimental/query-tests/Security/CWE/CWE-570/semmle/tests/WrongInDetectingAndHandlingMemoryAllocationErrors.expected

```diff
@@ -1,16 +1,14 @@
-| test.cpp:63:8:63:19 | call to operator new[] | memory allocation error check is incorrect or missing |
-| test.cpp:92:5:92:31 | call to operator new[] | memory allocation error check is incorrect or missing |
-| test.cpp:93:15:93:41 | call to operator new[] | memory allocation error check is incorrect or missing |
-| test.cpp:96:10:96:36 | call to operator new[] | memory allocation error check is incorrect or missing |
-| test.cpp:151:9:151:24 | call to operator new | memory allocation error check is incorrect or missing |
-| test.cpp:157:9:157:28 | call to operator new | memory allocation error check is incorrect or missing |
-| test.cpp:182:15:182:35 | call to operator new | memory allocation error check is incorrect or missing |
-| test.cpp:187:15:187:35 | call to operator new | memory allocation error check is incorrect or missing |
-| test.cpp:192:15:192:35 | call to operator new | memory allocation error check is incorrect or missing |
-| test.cpp:199:15:199:35 | call to operator new | memory allocation error check is incorrect or missing |
-| test.cpp:204:15:204:35 | call to operator new | memory allocation error check is incorrect or missing |
+| test.cpp:21:9:21:15 | new | This allocation cannot return null. $@ is unnecessary. | test.cpp:21:9:21:15 | new | This check |
+| test.cpp:29:13:29:24 | new[] | This allocation cannot return null. $@ is unnecessary. | test.cpp:30:7:30:13 | ... == ... | This check |
+| test.cpp:33:13:33:24 | new[] | This allocation cannot return null. $@ is unnecessary. | test.cpp:34:8:34:9 | p2 | This check |
+| test.cpp:37:13:37:24 | new[] | This allocation cannot return null. $@ is unnecessary. | test.cpp:38:7:38:16 | ... == ... | This check |
+| test.cpp:41:13:41:24 | new[] | This allocation cannot return null. $@ is unnecessary. | test.cpp:42:7:42:19 | ... == ... | This check |
+| test.cpp:45:13:45:24 | new[] | This allocation cannot return null. $@ is unnecessary. | test.cpp:46:7:46:8 | p5 | This check |
+| test.cpp:49:8:49:19 | new[] | This allocation cannot return null. $@ is unnecessary. | test.cpp:50:7:50:13 | ... == ... | This check |
+| test.cpp:53:8:53:19 | new[] | This allocation cannot return null. $@ is unnecessary. | test.cpp:54:8:54:9 | p7 | This check |
+| test.cpp:58:8:58:19 | new[] | This allocation cannot return null. $@ is unnecessary. | test.cpp:59:7:59:16 | ... == ... | This check |
+| test.cpp:63:8:63:19 | new[] | This allocation cannot return null. $@ is unnecessary. | test.cpp:64:7:64:19 | ... != ... | This check |
+| test.cpp:69:9:69:20 | new[] | This allocation cannot return null. $@ is unnecessary. | test.cpp:70:7:70:14 | ... != ... | This check |
+| test.cpp:75:11:75:22 | new[] | This allocation cannot return null. $@ is unnecessary. | test.cpp:76:13:76:15 | p11 | This check |
+| test.cpp:151:9:151:24 | new | This allocation cannot throw. $@ is unnecessary. | test.cpp:152:15:152:18 | { ... } | This catch block |
```

Comment:
We appear to have lost the results on lines 92, 93 and 96 but this is not acknowledged in the `test.cpp`.

Reply:
(I think it would be fair to rewrite the test case in question, or accept the loses as the cases are difficult to disentangle)

Reply:
Oh, that was certainly not my intention. Let me check why that is the case.

Reply:
Fixed in https://github.com/github/codeql/pull/5847/commits/95e65dec8faf1eb0836f09c3469a96a6fd1f8b1e.